docs/remote/devcontainer-cli.md

```diff
@@ -126,13 +126,87 @@ These steps above are also provided in the CLI repo's [README](https://github.co
 
 If you'd like to use the dev container CLI in your CI/CD builds or test automation, you can find examples of GitHub Actions and Azure DevOps Tasks in the [devcontainers/ci](https://github.com/devcontainers/ci) repository.
 
+## Pre-building
+
+The `devcontainer build` command allows you to build quickly build dev container image following the same steps the Remote - Containers extension or GitHub Codespaces will. This is particularly useful when you want to pre-build a dev container image using a CI or DevOps product like GitHub Actions.
+
+`build` accepts a path to the folder containing a `.devcontainer` folder or `.devcontainer.json` file. For example, `devcontainer build --workspace-folder <my_repo>` will build the container image for `my_repo`.
+
+### Example of building and publishing an image
+
+For example, you may want to pre-build a number of images that you then reuse across multiple projects or repositories. To do so, follow these steps:
+
+1. [Create](/docs/editor/versioncontrol.md#initialize-a-repository) a source code repository.
+
+1. Create dev container configuration for each image you want to pre-build, customizing as you wish (including [dev container Features](#dev-container-features-preview)). For example, consider this devcontainer.json file:
+
+    ```json
+    {
+        "build": {
+            "dockerfile": "Dockerfile"
+        },
+        "features": {
+            "docker-in-docker": "latest"
+        }
+    }
+    ```
+
+1. Use the `devcontainer build` command to build the image. See documentation for your image registry (like the [Azure Container Registry](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli), [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images), or [Docker Hub](https://docs.docker.com/engine/reference/commandline/push)) for information on image naming and additional steps like authentication.
+
+    ```bash
+    devcontainer build --workspace-folder <my_repo>
+    ```
+
+1. Next [push](https://docs.docker.com/engine/reference/commandline/push/) the image to your registry.
+
+    ```bash
+    docker push ghcr.io/your-org/your-image-name
```

Comment:
For this command, you can now use the `--push` flag in the `devcontainer build` step.

Reply:
Good call out! I'm trying the following, but it doesn't work for me - is my syntax incorrect?

```
devcontainer build --workspace-folder <my_repo> --push <my_org>/<my_image_name>:<optional_image_version>
```

So with a concrete repo as an example, I'm trying:
```
devcontainer build --workspace-folder vscode-remote-try-rust --push brmurtau/test-rust:latest
```

I'm able to do the following separate commands successfully, so wondering what I'm missing:
```
devcontainer build --workspace-folder vscode-remote-try-rust
docker tag vsc-vscode-remote-try-rust-<image version info> brmurtau/test-rust:latest
docker push brmurtau/test-rust:latest
```